### PR TITLE
feat(collector): add InfluxDB collector

### PR DESCRIPTION
**Pull request by affo**
# One-line summary

This PR adds the InfluxDB collector.

## Description

This PR adds the InfluxDB collector.

## Types of Changes
_What types of changes does your code introduce? Keep the ones that apply:_

- New feature (non-breaking change which adds functionality)

## Tasks
_List of tasks you will do to complete the PR_
  - [x] Implement collector
  - [x] Test in local env
  - [x] Add README

## Review
_List of tasks the reviewer must do to review the PR_
- [ ] Tests
- [ ] Documentation
- [ ] CHANGELOG

### CODE REVIEW

File: pkg/collector/influxdb_collector.go [hunk 1]
```diff
@@ -0,0 +1,150 @@
...
+	interval       time.Duration
+	metric         autoscalingv2.MetricIdentifier
+	metricType     autoscalingv2.MetricSourceType
+	query          string
+	orgID          string
+}
+
+func NewInfluxDBCollector(kubeClient kubernetes.Interface, hostPort string, token string, orgID string, hpa *autoscalingv2.HorizontalPodAutoscaler, config *MetricConfig, interval time.Duration) (*InfluxDBCollector, error) {
+	collector := &InfluxDBCollector{
+		kubeClient: kubeClient,
```
Comment (mikkeloscar) [inline pkg/collector/influxdb_collector.go]: the `kubeClient` doesn't seem to be in use anywhere?

File: pkg/server/start.go [hunk 2]
```diff
@@ -88,6 +88,12 @@ func NewCommandStartAdapterServer(stopCh <-chan struct{}) *cobra.Command {
 		"whether to enable External Metrics API")
 	flags.StringVar(&o.PrometheusServer, "prometheus-server", o.PrometheusServer, ""+
 		"url of prometheus server to query")
+	flags.StringVar(&o.InfluxDBHostPort, "influxdb-hostport", o.InfluxDBHostPort, ""+
```
Comment (mikkeloscar) [inline pkg/server/start.go]: Maybe a better name would be "address"?

**Comment by affo** [comment]
@mikkeloscar thank you for the quick walkthrough and for the comments.

I wanted to add the `work in progress` label to the PR, but I think I don't have permissions.

Anyways, I'll add the README section and address your comments asap.  
Once done that I'll ping you for further review. Sounds good?

Thanks again 👍

**Comment by mikkeloscar** [comment]
You can also prefix the title with `[WIP]` then it's at least clear to me that it's not ready for review.

**Comment by affo** [comment]
@mikkeloscar you are absolutely right  😅 Gotcha 👍

**Comment by affo** [comment]
@mikkeloscar I guess this is ready for review.

I addressed your comments and updated the README.

I also tested it in a local env and it runs fine. I was creating an example also to help you reproducing a  working example, but I deferred it to another PR if it could be useful 👍

**Comment by mikkeloscar** [comment]
Thanks, I'll get around to reviewing tomorrow or Friday. Please ensure you sign-off your commits so that check passes :)

**Comment by affo** [comment]
Yeah,  gonna squash and sign.

Is the coverage step an issue? Should I add more tests?

**Comment by mikkeloscar** [comment]
> Is the coverage step an issue? Should I add more tests?

I mean tests are always good to have :)

File: pkg/collector/influxdb_collector.go [hunk 3]
```diff
@@ -0,0 +1,152 @@
...
+			//  (by checking the message), but that seems quite dirty.
+			collector.query = query
+		} else {
+			return nil, fmt.Errorf("no Flux query defined for metric \"%s\"", config.Metric.Name)
+		}
+	default:
+		return nil, fmt.Errorf("unknown metric type: %v", configType)
+	}
+	// Use custom InfluxDB config if defined in HPA annotation.
+	if v, found := config.Config[influxDBAddressKey]; found {
```
Comment (szuecs) [inline pkg/collector/influxdb_collector.go]: `found -> ok`
Also the other ones, please :)

File: pkg/collector/influxdb_collector.go [hunk 4]
```diff
@@ -0,0 +1,152 @@
...
+		return nil, fmt.Errorf("InfluxDB does not support object, but only external custom metrics")
+	case autoscalingv2.ExternalMetricSourceType:
+		// `metricSelector` is flattened into the MetricConfig.Config.
+		queryName, ok := config.Config[influxDBQueryNameLabelKey]
+		if !ok {
+			return nil, fmt.Errorf("selector for Flux query is not specified,"+
+				"please add metricSelector.matchLabels.%s: <...> to .yml description", influxDBQueryNameLabelKey)
+		}
+		if query, ok := config.Config[queryName]; ok {
+			// TODO(affo): validate the query.
```
Comment (szuecs) [inline pkg/collector/influxdb_collector.go]: should this stay or do you want to validate it before merge?

Comment (affo) [reply pkg/collector/influxdb_collector.go]: This should stay, that's future work 👍

Comment (szuecs) [reply pkg/collector/influxdb_collector.go]: ok, maybe better to add an issue and reference it from the code?

**Comment by affo** [comment]
@szuecs addressed your comments.

@mikkeloscar added tests for collector instantiation.

Thank you for reviewing.

**Comment by szuecs** [comment]
:+1:

**Comment by mikkeloscar** [comment]
:+1:

**Pull request merged**